model/task/task.go

```diff
@@ -2486,82 +2487,89 @@ func (t *Task) Insert() error {
 // are also archived.
 func (t *Task) Archive() error {
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+		err := ArchiveMany([]Task{*t})
+		return errors.Wrapf(err, "archiving display task '%s'", t.Id)
+	} else {
+		// Archiving a single task.
+		archiveTask := t.makeArchivedTask()
+		err := db.Insert(OldCollection, archiveTask)
 		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
+			grip.Error(message.WrapError(err, message.Fields{
+				"archive_task_id": archiveTask.Id,
+				"old_task_id":     archiveTask.OldTaskId,
+				"execution":       t.Execution,
+				"display_only":    t.DisplayOnly,
+			}))
+			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		err = UpdateOne(
+			bson.M{IdKey: t.Id},
+			bson.M{
+				"$unset": bson.M{
+					AbortedKey:              "",
+					AbortInfoKey:            "",
+					OverrideDependenciesKey: "",
+				},
+				"$set": bson.M{
+					ExecutionKey: t.Execution + 1,
+				},
+			})
+		if err != nil {
+			return errors.Wrap(err, "updating task")
 		}
+		t.Aborted = false
+		return nil
 	}
+}
 
-	archiveTask := t.makeArchivedTask()
-	err := db.Insert(OldCollection, archiveTask)
-	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"archive_task_id": archiveTask.Id,
-			"old_task_id":     archiveTask.OldTaskId,
-			"execution":       t.Execution,
-			"display_only":    t.DisplayOnly,
-		}))
-		return errors.Wrap(err, "inserting archived task into old tasks")
-	}
-	err = UpdateOne(
-		bson.M{IdKey: t.Id},
-		bson.M{
-			"$unset": bson.M{
-				AbortedKey:              "",
-				AbortInfoKey:            "",
-				OverrideDependenciesKey: "",
-			},
-			"$inc": bson.M{ExecutionKey: 1},
-		})
-	if err != nil {
-		return errors.Wrap(err, "updating task")
-	}
-	t.Aborted = false
-	return nil
+func shouldRestartTask(resetFailedWhenFinished bool, resetWhenFinished bool, status string) bool {
+	return resetWhenFinished || !resetFailedWhenFinished || evergreen.IsFailedTaskStatus(status)
 }
 
+// ArchiveMany accepts tasks and display tasks (no execution tasks), bundles them all up, archives and updates all
+// that need to be (accounting for ResetFailedWhenFinished) in two queries
 func ArchiveMany(tasks []Task) error {
-	if len(tasks) == 0 {
-		return nil
-	}
-	existingTasksMap := map[string]bool{}
-	for _, t := range tasks {
-		existingTasksMap[t.Id] = true
-	}
-	additionalTasks := []string{}
+	bundledTasks := []string{}        // Contains all tasks and display tasks
```

Comment:
allTasks?

Reply:
Or is this toArchiveIds ? I think it's helpful for variable names to match each other better if they contain similar information, and for the comment / variable name to say Id if it's just Ids (like at 2534)

Reply:
I clarrified it and it should be more apparent (please let me know!!) but the bundledTasks is only tasks and display tasks Ids.

All tasks and display tasks that are put in to archivemany will always be archived